File: src/main/java/me/santio/uhc/scenarios/DoubleDropsScenario.java
package me.santio.uhc.scenarios;

import me.santio.uhc.exceptions.DuplicateScenarioException;
import me.santio.uhc.models.Scenario;
import me.santio.uhc.models.ScenarioData;
import org.bukkit.Material;
import org.bukkit.World;
import org.bukkit.event.EventHandler;
import org.bukkit.event.block.BlockBreakEvent;
import org.bukkit.inventory.ItemStack;

public class DoubleDropsScenario extends Scenario {
    public DoubleDropsScenario() throws DuplicateScenarioException {
        super("Double Drops");
        setIcon(Material.COAL_ORE);
        createData(new ScenarioData("all_blocks", "Should all blocks be affected instead of just ores", false));
    }
    
    @EventHandler
    public void onMineOre(BlockBreakEvent event) {
        if (event.isCancelled()) return;
        if (!(boolean) getData().get("all_blocks").getValue() && !event.getBlock().getType().toString().contains("_ORE")) return;
        
        World world = event.getBlock().getWorld();
        event.getBlock().getDrops(event.getPlayer().getInventory().getItemInMainHand(), event.getPlayer()).forEach((ItemStack item) -> world.dropItem(event.getBlock().getLocation(), item));
    }
}

File: src/main/kotlin/me/santio/uhc/scenarios/DoubleDropsScenario.kt
package me.santio.uhc.scenarios

import me.santio.uhc.models.Scenario
import org.bukkit.Material
import org.bukkit.event.EventHandler
import org.bukkit.event.block.BlockBreakEvent
import org.bukkit.inventory.ItemStack
import java.util.function.Consumer

class DoubleDropsScenario : Scenario("Double Drops") {
    init {
        icon = Material.COAL_ORE
//        createData(ScenarioData("all_blocks", "Should all blocks be affected instead of just ores", false))
    }

    @EventHandler
    fun onMineOre(event: BlockBreakEvent) {
        if (event.isCancelled) return
//        if (!getData().get("all_blocks").getValue() && !event.block.type.toString().contains("_ORE")) return
        val world = event.block.world
        event.block.getDrops(event.player.inventory.itemInMainHand, event.player)
            .forEach(Consumer { item: ItemStack? -> world.dropItem(event.block.location, item!!) })
    }
}